src/BP-CoreLib/OptionMenu/ActionLabel.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using BPCoreLib.Interfaces;
+using BrokeProtocol.API;
+using BrokeProtocol.Collections;
+using BrokeProtocol.Entities;
+using BrokeProtocol.Required;
+
+namespace BPCoreLib.OptionMenu
+{
+    public class ActionLabel : IActionLabel
+    {
+        public string Name { get; }
+        public Action<ShPlayer, string> Action { get; }
+
+        public ActionLabel(string name, Action<ShPlayer, string> action)
+        {
+            this.Name = name;
+            this.Action = action;
+        }
+    }
+
+    internal class OptionMenu
+    {
+        private readonly string _title;
+        public Dictionary<string, ActionLabel> Actions { get; private set; }
+        private readonly LabelID[] _options;
+        private readonly string menuId;
```

Comment:
Inconsistent field naming. private readonly fields should be camelCased, prefixed with a `_`.
```suggestion
        private readonly string _title;
        public Dictionary<string, ActionLabel> Actions { get; private set; }
        private readonly LabelID[] _options;
        private readonly string _menuId;
```